packages/scene-composer/src/components/toolbars/common/ToolbarItem.tsx

```diff
@@ -29,12 +29,56 @@ export function ToolbarItem<T extends ToolbarItemOptions>({
   const [selectedItem, setSelectedItem] = useState(initialSelectedItem ?? items[0]);
   const [showMenu, setShowMenu] = useState<boolean>();
   const itemContainerRef = useRef<HTMLDivElement | null>(null);
+  // refs used to trap focus on menu items
+  const firstItemRef = useRef<HTMLDivElement | null>(null);
+  const lastItemRef = useRef<HTMLDivElement | null>(null);
+
+  const isItemFirst = (index: number) => {
+    return (type === 'action-select' && index === 1) || (type !== 'action-select' && index === 0);
+  };
+
+  const isItemLast = (index: number) => {
+    return index === items.length - 1;
+  };
+
+  const getItemRef = (index: number) => {
+    if (isItemFirst(index)) {
+      return firstItemRef;
+    } else if (isItemLast(index)) {
+      return lastItemRef;
+    }
+    return null;
+  };
 
   function handleItemClick(item: T) {
     type === 'mode-select' && setSelectedItem(item);
-    onClick?.(item);
+    onSelect?.(item);
   }
 
+  const handleItemKeyDown = (item: T, e, focusIndex?: number) => {
```

Comment:
Should this be a useCallback since we are passing it down? Could very well not be, I'm no expert on useCallbacks

Reply:
Yeah, since the component it's being passed to is memoized we can wrap this in useCallback -- same with `handleItemClick` above